kernel/kernel-api/src/main/java/io/delta/kernel/internal/replay/ConflictChecker.java

```diff
@@ -88,10 +92,13 @@ public static TransactionRebaseState resolveConflicts(
 
     public TransactionRebaseState resolveConflicts(Engine engine) throws ConcurrentWriteException {
         List<FileStatus> winningCommits = getWinningCommitFiles(engine);
+        AtomicReference<Optional<CommitInfo>> winningCommitInfoOpt =
```

Comment:
why is this a atmoic reference? Why can't we just use the Optional?

Reply:
Because of this error:
https://github.com/delta-io/delta/pull/3283#discussion_r1645254811